## Практическое задание N1

### Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

In [37]:
from math import factorial

In [38]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

> а) Общее число событий С4:52

In [39]:
C4_52 = combinations(52, 4)
C4_52

270725

>Т23456789 10 ВДК - 13 карт одной масти (крестей)
>
>Количесво благоприятный исходов C4:13
>

In [40]:
C4_13 = combinations(13, 4)
C4_13

715

In [41]:
P = C4_13 / C4_52
P

0.0026410564225690276

> Вероятность P=0.002641

> b) Общее число событий C4:52

In [42]:
C4_52 = combinations(52, 4)
C4_52

270725

> Число событий где тузы не попали.

> Вынимали 4 карты там нет тузов 52-4 туза = 48

> С4:48

In [43]:
C4_48 = combinations(48, 4)
C4_48

194580

> Число благоприятных событий (с тузами) = из общего числа комбинаций где вытаскивали 4 карты вычесть комбинации без тузов

In [44]:
C4ace = C4_52 - C4_48
C4ace

76145

> Вычисляем вероятность

In [45]:
P = C4ace / C4_52
P

0.2812632745405855

> Вероятность P=0.28

2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

> Всего вариантов C3:10

In [46]:
C3_10 = combinations(10, 3)
C3_10

120

> Благоприятных комбинаций

In [47]:
C3_3 = combinations(3, 3)
C3_3

1

In [48]:
P = C3_3/C3_10
P

0.008333333333333333

> Вероятность P = 0.0083
>

3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?
> Через условную вероятность


In [49]:
mark1 = 9/15
mark2 = 8/14
mark3 = 7/13
P = mark1 * mark2 * mark3
P

0.18461538461538457

> Вероятность P = 0.1846

4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?


> Общее число вариантов С2:100

In [50]:
C2_100 = combinations(100,2)
P = 1/C2_100
P

0.00020202020202020202

> Вероятность P = 0.0002
